### ISSUE TITLE
tests: integrate aXe (accessibility engine)

### ISSUE THREAD

**Issue by garrett** [cockpit-project/cockpit#8641]
aXe checks HTML + CSS for several accessibility-related issues and reports the status:
https://www.axe-core.org/

It can be ran in-browser, but also is able to be integrated into test suites.

- https://www.axe-core.org/examples/
- https://www.axe-core.org/examples/phantomjs.html

Accessibility checks should be run on every pull request.

We should also generate a list of errors across Cockpit master of issues to squash as well. These should probably be "grandparented in" for a few months as we fix them so we're not triggering errors in the meantime.

(However, if new code doesn't pass aXe, that should be considered an error and must be fixed before it is accepted.)

**Comment by garrett** [comment]
I recently found some additional React-specific accessibility testing tools too:

- [eslint-plugin-jsx-a11y](https://github.com/evcohen/eslint-plugin-jsx-a11y): Static AST checker for a11y rules on JSX elements
- [react-axe](https://github.com/dequelabs/react-axe): Accessibility auditing for React.js applications
- [react-a11y](https://github.com/reactjs/react-a11y): Identifies accessibility issues in your React.js elements
- [axe-webdriverjs](https://github.com/dequelabs/axe-webdriverjs): Provides a chainable aXe API for Selenium's WebDriverJS and automatically injects into all frames
  - [react-axe-webdriverjs-demo](https://github.com/marcysutton/react-axe-webdriverjs-demo): A demonstration of using axe-webdriverjs with a React application

**Comment by garrett** [comment]
I'm happy to assist where I can, but I don't understand how our testing framework works and can't integrate it by myself. :cry:
